src/features/card/actions/ui/update-changes.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { button } from '@box/ui';
+import { useEvent, useStore } from 'effector-react/ssr';
+
+import * as model from '../model';
+
+/**
+ * Сохранение изменений по карточке (edit)
+ * @remark Сброс до изначального состояния модели
+ */
+export const UpdateChanges = () => {
+  const draft = useStore(model.$draft);
+  const submitChanges = useEvent(model.submitChanges);
+
+  return (
+    <button.Primary onClick={() => submitChanges(draft)} disabled={!draft}>
+      Save
+    </button.Primary>
+  );
+};
```

Comment:
suggestion(non-blocking): Можно переписать на reflect:

```suggestion
export const UpdateChanges = reflect({
  view: button.Primary,
  bind: {
    onClick: submitChangesPressed,
    disabled: $draft.map(draft => !draft),
    children: 'Save',
  },
})
```

issue: Здесь данные из стора передаются в ивент во вью, а это часть логики.
suggestion: Можно запросто взять данные из draft в момент вызова submitChanges, например через sample. Тогда вьюхе это просто не нужно будет знать.

issue: Аналогично с disabled. Причина по которой кнопка должна отключаться это тоже часть модели.
suggestion: Лучше из модели экспортировать соответствующий стор `$disabled`

```suggestion
export const UpdateChanges = reflect({
  view: button.Primary,
  bind: {
    onClick: submitChangesPressed,
    disabled: $isSubmitDisabled,
    children: 'Save',
  },
})
```

nitpick: Я не особо вижу смысла выносить такой простейший компонент в фичу, тогда как он решается одним простейшим reflect и он напрямую прибит к логике и фактически не является реюзабельным

Reply:
>  Можно переписать на reflect

Переписать то можно, а какой профит?)

Если захочется потом что-то поправить в логике,то придется шаманить с этими reflect-декларациями, и то скорей всего придется явно объявлять компонент, т.к. у нас не будет доступа к самой разметке и логике компонента

ИМХО, здесь излишне)

Reply:
> Можно запросто взять данные из draft в момент вызова submitChanges, например через sample. Тогда вьюхе это просто не нужно будет знать.

Да, мне тож моя версия не особо заходит, но там с аттачем стора в модели почему-то не работало

Но пробнем еще раз значит ~(с v0 же не спешим :D)~~

Reply:
> Лучше из модели экспортировать соответствующий стор $disabled

Оки, согл

Reply:
>  Я не особо вижу смысла выносить такой простейший компонент в фичу, тогда как он решается одним простейшим reflect и он напрямую прибит к логике и фактически не является реюзабельным

Мне кажется, или с таким подходом можно вообще тогда забить на слои и делать все на уровне страниц, которые будут разбухать?))

**Мыж вроде за тонкие страницы были**

**И при этом, чтобы логика была расфасована по скоупа ответственности (тобишь по слоям)**

Reply:
Рефлект нужен, чтобы покрывать как раз такие кейсы. если сделано не на нем это сразу бросается в глаза, мол «смотри, тут не на рефлекте, значит что-то особенное и не тривиальная логика отображения»

Reply:
> Мыж вроде за тонкие страницы были
Не стоит впадать в крайность, так можно и каждый стор с ивентом вынести

кнопка отправки это просто кнопка, здесь нет никакой логики

Reply:
Нуууу, оке

Но при этом, опять же, не вижу особо проблемы, чтобы скоуп работы с изменениями (сейчас edit, но и create сюда же попадет потом) отделять от обычной логики страницы (тобишь выносить в другой слой)

У нас как раз есть `feature/card-actions`, где хранятся базовые контролы для "редактирования полей" карточки

Reply:
Я топлю за удаление фичи card-actions. Она неправильно сформулирована.
У тебя будет card-draft, а остальная логика в странице

Reply:
`question:` Ну т.е. ты от `features/card-draft` предлагаешь оставить по большей части тольк модель саму? (или вообще вынести все в entities такое?)

И при этом кастовать жирные рефлекты что на странице EditPage, что на странице CreatePage?

> Спрашиваю без наброса, просто пока рили кажется, что от такого лишь размажется скоуп ответственности и сами страницы разбухнут (а ты сам побольше моего топил за то, чтобы на страницах не держать лишней логики)

Reply:
Все поправил, но рефлект пока не буду юзать

ИМХО, преждевременная оптимизация, тем более это все в фиче регулируется а не в компоненте страницы

А то что это фича все же - обсудили уже в фиче слайсах